apis/core/v1alpha1/resource.go

```diff
@@ -208,3 +208,23 @@ type ProviderSpec struct {
 	// the credentials that are used to connect to the provider.
 	CredentialsSecretRef SecretKeySelector `json:"credentialsSecretRef"`
 }
+
+// A TargetSpec defines the common fields of objects that satisfy the Target interface.
+type TargetSpec struct {
+	// SecretRef specifies the name of a Secret, in the same namespace as this
+	// target, to which any connection details for this target should be written
+	// or already exist. Connection secrets referenced by a target should
+	// contain information for connecting to a resource that allows for
+	// scheduling of workloads. +optional
+	SecretRef *LocalSecretReference `json:"secretRef,omitempty"`
```

Comment:
Opted for `SecretRef` instead of `ConnectionSecretRef` because it seems implied that the the secret is will be for connection in this case. However, maybe it would be good to be more verbose?

Reply:
Agreed, probably will change this to `ConnectrionSecretRef`.